packages/app_config/lib/app_config.dart

```diff
@@ -0,0 +1,24 @@
+library app_config;
+
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'config.dart';
+
+class AppConfig extends InheritedWidget {
+  final String title;
+
+  AppConfig({
+    required FlavorConfig data,
+    required super.child,
+    super.key,
+  }) : title = data.appTitle {
+    GetIt.I.registerSingleton<FlavorConfig>(data);
+  }
+
+  static AppConfig? of(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType(aspect: AppConfig);
+  }
+
+  @override
+  bool updateShouldNotify(covariant InheritedWidget oldWidget) => false;
```

Comment:
Напишу на русском ибо много букафф. Я бы сделал это по другому, создал бы в папке либ папку с флейварами, перенес бы файл FlavorConfig туда и прокидывал бы FlavorConfig в di, так бы ты смог:
1) Обойтись без инхирита
2) Сразу все засетать в di(в том плане, что ты сразу передашь baseurl для dio  и тд)
3) Если тебе надо менять ui часть с помощью конфига, то я бы передал все что надо для ui в параметры конструктора StarMovieApp, а дальше создал бы какой-то сингалтон типа AppScreenConfig(это если тебе надо будет на каждом скрине брать что-то из конфига) и туда все передал, что прилетает из конструктора.
Но наверное как сделать более правильно подскажут либо @sergei-mikhailovskii-idf  либо @yuliya-klescheva-idf 
Но у тебя тоже интересное решение )

Reply:
Дело в том, что я тоже думал сразу закинуть в ДИ, но решил поискать альтернативу и вот что вышло =) Оке, давай узнаем что порекомендуют сергей и юля)

Reply:
I don't see any reason to keep the inherited widget here. it is just the POJO model (in our case PODO :) ), which represents a json file. so, in my view u should have the following flow

Reply:
<img width="626" alt="image" src="https://user-images.githubusercontent.com/64923802/193307503-462def5e-58d9-41a9-b9df-7b7ece492a7e.png">

its logic seems to be transparent

Reply:
about inherited widget, in our case this may be redundant but in perspective it more preferable then singleton, on my opinion)

Reply:
maybe. let's assume, we have the following situation - we have screens A, B, C, D. They are opened one by one (click smth at A - opens B. click smth at B - opens C and so on). at D screen we need to execute the api call only in case we have prod flavor. in case without inhereted widget we will do the following:

1) read the config in root lib folder
2) pass it to di container
3) pass it to presentation di
4) pass it to bloc of screen D
5) add check to the bloc

```dart
if (config.flavor == prod) {
    useCase();
}
```

in case with inhereted widget u will have the following flow:
1) read the config in root lib folder
2) wrap application/some widget inside with inhereted widget
3) get the inhereted value at screen D
4) pass it to bloc

yes, one step shorter, but seems that both variants are ok

the problem which I see in inhereted widget, it seems strange at the data layer, because it's widget (presentation part)

P.S. it's not obvious to register the config in di